From 9220f76b17816b476f7662bae959d9f403046a37 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sat, 18 Nov 2017 18:32:48 +0100
Subject: [PATCH] Remove ThreadPool.h inclusion in upnpdebug.h

ThreadPool.h is now a private header which is not installed so remove
this header from upnpdebug.h otherwise an external application which
includes upnpdebug.h will crash

Also add LinkedList.h to service_table.h as it was included by
ThreadPool.h in upnpdebug.h

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 upnp/inc/upnpdebug.h         | 1 -
 upnp/src/inc/service_table.h | 1 +
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/upnp/inc/upnpdebug.h b/upnp/inc/upnpdebug.h
index 45a86bbb..fc31df5a 100644
--- a/upnp/inc/upnpdebug.h
+++ b/upnp/inc/upnpdebug.h
@@ -37,7 +37,6 @@
  * \file
  */
 
-#include "ThreadPool.h"
 #include "upnpconfig.h"
 #include "UpnpGlobal.h"		/* for UPNP_INLINE */
 
diff --git a/upnp/src/inc/service_table.h b/upnp/src/inc/service_table.h
index e7b6d7b8..dfd58b75 100644
--- a/upnp/src/inc/service_table.h
+++ b/upnp/src/inc/service_table.h
@@ -46,6 +46,7 @@ extern "C" {
 #include "ixml.h"
 #include "upnp.h"
 #include "upnpdebug.h"
+#include "LinkedList.h"
 
 #include <stdio.h>
 #include <time.h>
